data-platform/core-converged-db/oracle-database-23c-new-features/OracleJDBC23cNewFeatures/src/main/java/swierzy/oraclejdbc23newfeatures/Main.java

```diff
@@ -0,0 +1,156 @@
+
+        psQuery = con.prepareStatement("SELECT COUNT(*) FROM TEST");
+
+        /*
+           INSERT and SELECT statements reactive calls
+           notes:
+           1. reactive methods are implemented ONLY as Oracle extensions to the standard JDBC interfaces
+              - it is not possible to call them without using OraclePrepareStatement class directly
+           2. traditional JAVA casting due to Oracle official documentation is not recommended
+              - it is needed to use class.unwrap method to get the OraclePreparedStatement object
+         */
+        Flow.Publisher<OracleResultSet> fpQuery =  psQuery.unwrap(OraclePreparedStatement.class).executeQueryAsyncOracle();
+        fpQuery.subscribe(new QuerySubscriber(latch));
+        System.out.println("Main.reactiveCallsDemo: SELECT started in NOWAIT mode");
+
+        Flow.Publisher<Long> fpDML = psDML.unwrap(OraclePreparedStatement.class).executeUpdateAsyncOracle();
+        fpDML.subscribe(new DMLSubscriber(latch));
+        System.out.println("Main.reactiveCallsDemo: INSERT started in NOWAIT mode");
+        /*
+           As this demo uses reactive methods, the execution does not wait for the end of SQL statements
+           reactiveCallsDemo() execution is completed BEFORE completion of INSERT and SELECT statements reactive execution
+         */
+        System.out.println("Main.reactiveCallsDemo: end of method and return to Main.main.");
+
+    }
+
+    /*
+       stramsIngestionDemo(): method, which demonstrates support for Streams Ingestion
+       provided by Oracle 23c JDBC driver
+     */
+    public static void streamsIngestionDemo() throws Exception {
+        String value;
+
+        // auxiliary, traditional database connection used to check number of rows in TEST table
+        Connection conAux = DriverManager.getConnection(
+                "<db_url>",
+                "<db_username>",
+                "<db_password>");
+
+        // auxiliary PreparedStatement object used to check number of rows in TEST table
+        PreparedStatement psAux = conAux.prepareStatement("SELECT COUNT(*) FROM TEST");
+        // auxiliary ResultSet object used to check number of rows in TEST table
+        ResultSet rsAux;
+        int countInt;
+
+        /*
+           Streams Ingestion additionally to traditional JDBC parameters, like
+           connection string, username and password, requires providing the following data
+           1. ExecutorService: pool of threads responsible for execution
+           2. buffer size
+           3. interval (in seconds) between pushing the data into the database
+           4. names of target schema, table and columns
+         */
+
+        System.out.println("Main.streamsIngestDemo: connecting to the database.");
+        ExecutorService es = Executors.newFixedThreadPool(5);
+        ReactiveStreamsIngestion rs = ReactiveStreamsIngestion
+                .builder()
+                .url("<db_url>")
```

Comment:
Don't repeat yourself and make it more flexible for the reader. Get these values from parameters.